### PR TITLE
[#14] [UI] As a user, I can see the Home loading screen

### LINKED ISSUE #14: [UI] As a user, I can see the Home loading screen
## Why

While the app is downloading the survey list, users should see the skeleton loading animation on Home screen

## Acceptance Criteria

- Loading screen must display after user logged in, following the design on Figma

## Design

Screenshot 2023-03-13 at 6 02 37 PM

### PR DESCRIPTION
**Pull request by ducbm051291**
- Close #14 

## What happened 👀

- Add Shimmer package in pubspec
- Create HomeLoading, HomeLoadingText
- Integrate HomeLoading in the Home screen

## Insight 📝

- N/A

## Proof Of Work 📹

CleanShot 2023-04-28 at 15 33 34

### CODE REVIEW

File: lib/ui/home/loading/home_loading.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        ],
+      ),
+    );
+  }
+
+  Widget _buildHeaderLoading() => Container(
+        padding: const EdgeInsets.only(
+          top: 10,
+          right: 20,
+          bottom: 10,
```
Comment (thiennguyen0196) [inline lib/ui/home/loading/home_loading.dart:28]: Please use dimension value here for reuseable code
```suggestion
          top: Dimensions.paddingSmall,
          right: Dimensions.paddingMedium,
          bottom: Dimensions.paddingSmall,
```

File: lib/ui/home/loading/home_loading.dart [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            const HomeLoadingText(width: 40.0),
+            const SizedBox(height: 16.0),
+            HomeLoadingText(width: MediaQuery.of(context).size.width - 102.0),
+            const SizedBox(height: 4.0),
+            HomeLoadingText(width: MediaQuery.of(context).size.width - 240.0),
+            const SizedBox(height: 16.0),
+            HomeLoadingText(width: MediaQuery.of(context).size.width - 36.0),
+            const SizedBox(height: 4.0),
+            HomeLoadingText(width: MediaQuery.of(context).size.width - 140.0),
+            const SizedBox(height: 16.0)
```
Comment (thiennguyen0196) [inline lib/ui/home/loading/home_loading.dart:69]: Ditto here
```suggestion
            const SizedBox(height: Dimensions.paddingNormal),
            HomeLoadingText(width: MediaQuery.of(context).size.width - 102.0),
            const SizedBox(height: 4.0),
            HomeLoadingText(width: MediaQuery.of(context).size.width - 240.0),
            const SizedBox(height: Dimensions.paddingNormal),
            HomeLoadingText(width: MediaQuery.of(context).size.width - 36.0),
            const SizedBox(height: 4.0),
            HomeLoadingText(width: MediaQuery.of(context).size.width - 140.0),
            const SizedBox(height: Dimensions.paddingNormal)
```

File: lib/ui/home/loading/home_loading_text.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  const HomeLoadingText({
+    Key? key,
+    required this.width,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16.0),
```
Comment (thiennguyen0196) [inline lib/ui/home/loading/home_loading_text.dart:14]: Ditto here
```suggestion
      padding: const EdgeInsets.symmetric(horizontal: Dimensions.paddingNormal),
```

File: lib/utils/app_route.dart [hunk 4]
```diff
@@ -38,7 +38,7 @@ class AppRouter {
           if (!isLogin &&
               subLocation != RoutePath.forgotPassword.path &&
               subLocation != RoutePath.login.path) {
-            return RoutePath.login.path;
+            return RoutePath.home.path;
```
Comment (thiennguyen0196) [inline lib/utils/app_route.dart:41]: is this the test code @ducbm051291 😄

**Review by thiennguyen0196 — COMMENTED** [review]
Some concerns from me about the dimension values, the rest LGTM

File: lib/utils/app_route.dart (hunk 4 above)
Comment (ducbm051291) [reply lib/utils/app_route.dart:41]: Yes

**Review by nmint8m — APPROVED** [review]
Look good to me

**Review by thiennguyen0196 — APPROVED** [review]
lgtm

**Pull request merged**